src/Bicep.Core/Workspaces/Workspace.cs

```diff
@@ -26,6 +29,28 @@ public IEnumerable<ISourceFile> GetSourceFilesForDirectory(Uri fileUri)
         public ImmutableDictionary<Uri, ISourceFile> GetActiveSourceFilesByUri()
             => activeFiles.ToImmutableDictionary();
 
+        public void UpsertBicepConfig(Uri uri, BicepConfig bicepConfig)
+        {
+            activeBicepConfigs.AddOrUpdate(
+                uri,
+                (uri) => bicepConfig,
+                (uri, prevConfig) => bicepConfig);
+        }
+
+        public BicepConfig? GetBicepConfig(Uri uri)
+        {
+            activeBicepConfigs.TryGetValue(uri, out BicepConfig? bicepConfig);
+            return bicepConfig;
+        }
+
+        public void RemoveBicepConfig(Uri uri)
+        {
+            if (activeBicepConfigs.ContainsKey(uri))
+            {
+                activeBicepConfigs.TryRemove(uri, out _);
+            }
```

Comment:
You don't need the `ContainsKey` check - `TryRemove` does this.

Reply:
Removed here:
https://github.com/Azure/bicep/commit/bbb280e8f2fe05d4c7000044b10e5f9715fcb854